src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static partial class OperationExtensions
+    {
+        public static bool IsTargetOfObjectMemberInitializer(this IOperation operation)
+            => operation.Parent is IAssignmentOperation assignmentOperation &&
+               assignmentOperation.Target == operation &&
+               assignmentOperation.Parent?.Kind == OperationKind.ObjectOrCollectionInitializer;
+
+        /// <summary>
+        /// Returns the <see cref="ValueUsageInfo"/> for the given operation.
+        /// This extension can be removed once https://github.com/dotnet/roslyn/issues/25057 is implemented.
+        /// </summary>
+        public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
+        {
+            /*
+            |    code         | Read | Write | ReadableRef | WritableRef | NonReadWriteRef |
+            | x.Prop = 1      |      |  ✔️   |             |             |                 |
+            | x.Prop += 1     |  ✔️  |  ✔️   |             |             |                 |
+            | x.Prop++        |  ✔️  |  ✔️   |             |             |                 |
+            | Foo(x.Prop)     |  ✔️  |       |             |             |                 |
+            | Foo(x.Prop),    |      |       |     ✔️      |             |                 |
+               where void Foo(in T v)
```

Comment:
what if the `Prop` returns by value and not by ref, in which case there is a copy made? and other cases where there is a copy instead of a reference when `in` is skipped? is this handled properly?